server/server_app.js

```diff
@@ -176,15 +178,72 @@ app.get('/api/mail/verification', async (req, res) => {
 })
 
 /**
- * Get request accessing to the user profile.
+ * Get request to confirm a custom action.
+ * Link sended by e-mail
+ * Delete -> Remove the user credentials from the database
+ */
+app.get('/api/mail/customVerification', async (req, res) => {
+  const token = req.query.token
+  try {
+    const decoded = jwt.decode(token, process.env.JWT_SECRET)
+    const user = await database.prisma.User.findUnique({
+      where: {
+        id: decoded.id
+      }
+    })
+    const process = user.confirmProcess
+    await database.prisma.User.update({
+      where: {
+        id: decoded.id
+      },
+      data: {
+        confirmProcess: ''
+      }
+    })
+    if (process == 'Delete') {
+      await database.prisma.User.delete({
+        where: {
+          id: decoded.id
+        }
+      })
+    }
+    res.send('Operation ' + process + ' authorized and executed.')
+  } catch (err) {
+    console.error(err.message)
+    res.status(401).send('No matching user found.')
+  }
+})
+
+/**
+ * Get request to delete an account
+ * Send a confirmation e-mail before deleting.
  * Need to be authentified with a token.
  */
 app.get(
-  '/profile',
+  '/api/user/deleteAccount',
```

Comment:
is the authentification service part of the Api ?

Reply:
Yes, all user operation is a path from the API cause it interact with the Database.
But we will rebase it later with a more formal path norme. (Cf the list you made last week) ;)